priam/src/main/java/com/netflix/priam/cluster/management/Compaction.java

```diff
@@ -0,0 +1,169 @@
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.priam.cluster.management;
+
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.defaultimpl.CassandraOperations;
+import com.netflix.priam.merics.CompactionMeasurement;
+import com.netflix.priam.merics.IMetricPublisher;
+import com.netflix.priam.scheduler.CronTimer;
+import com.netflix.priam.scheduler.TaskTimer;
+import com.netflix.priam.utils.JMXConnectorMgr;
+import org.apache.commons.lang3.StringUtils;
+import org.quartz.CronExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to compact the keyspaces/columnfamilies
+ * Created by aagrawal on 1/25/18.
+ */
+@Singleton
+public class Compaction extends IClusterManagement<String> {
+    private static final Logger logger = LoggerFactory.getLogger(Compaction.class);
+    private final IConfiguration config;
+    private static final Pattern columnFamilyFilterPattern = Pattern.compile(".\\..");
+    private final CassandraOperations cassandraOperations;
+
+    @Inject
+    public Compaction(IConfiguration config, IMetricPublisher metricPublisher, CassandraOperations cassandraOperations) {
+        super(config, Task.COMPACTION, metricPublisher, new CompactionMeasurement());
+        this.config = config;
+        this.cassandraOperations = cassandraOperations;
+    }
+
+    public final Map<String, List<String>> updateCompactionCFList(IConfiguration config) throws Exception {
+        final Map<String, List<String>> columnFamilyFilter = new HashMap<>(); //key: keyspace, value: a list of CFs within the keyspace
+
+        if (config.getCompactionCFList() == null || config.getCompactionCFList().isEmpty()) {
+            logger.info("Compaction: No override provided by user. All keyspaces qualify for compaction.");
+
+            if (cassandraOperations != null)
+                cassandraOperations.getKeyspaces().forEach(keyspaceName -> {
+                List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());
+                columnFamilyFilter.put(keyspaceName, existingCfs);
+            });
+            return columnFamilyFilter;
+        }
+
+        logger.info("Compaction: Override provided by user. Calculating KS/CF's to qualify for compaction");
+        String[] filters = config.getCompactionCFList().split(",");
```

Comment:
When CFList is such that all CFs in keyspace are included, it would be better if you let C* decide the parallelism based on its concurrent compactors, so that compactions would go faster, to achive that, you can just issue keyspace level compaction and let C* do the job.

Reply:
Actually it is already there, if you see line 87 (I added comment there)